src/harness/reference_models/move_list_calc/move_list.py

```diff
@@ -0,0 +1,435 @@
+    interf_list = []
+    for cbsd_grant in cbsd_grant_list:
+        interf = computeInterference(cbsd_grant, constraint, num_iter)
+        interf_list.append(interf)
+
+    # Sort grants by their median interference contribution, smallest to largest
+    interf_list.sort(key=itemgetter('medianInterference'))
+
+    # Get indices, bearings, interference contributions of the sorted grants
+    sorted_grant_index = [i.get('grantIndex') for i in interf_list]
+    sorted_bearing = [i.get('bearing_c_cbsd') for i in interf_list]
+    I = np.array([i.get('randomInterference') for i in interf_list]).transpose()
+
+    return I, sorted_grant_index, sorted_bearing
+
+
+def grantsInMainBeam(ant_az, bearing):
+
+    """
+    Identify the grants in the main beam of the receiver antenna.
+
+    Inputs:
+        ant_az:     antenna azimuth (degrees)
+        bearing:    a list of bearing values (due north) from the protection point
+                    to the CBSDs
+
+    Returns:
+        an array contains index of CBSD inside the main beam
+    """
+
+    idx = []
+
+    # Compute off-axis angle between the axis of the main-beam of the incumbent
+    # receiver and the line between the protection point and each CBSD
+    for i in range(len(bearing)):
+        theta = ant_az - bearing[i]
+        if theta > 180:
+            theta -= 360
+        elif theta < -180:
+           theta += 360
+
+        # Check if the CBSD is inside or outside the antenna main-beam
+        if (theta >= (-1) * RADAR_BEAMWIDTH/2.0) and (theta <= RADAR_BEAMWIDTH/2.0):
+            idx.append(i)
+
+    return np.array(idx)	
+
+
+def find_nc(I, bearing, t):
+
+    """
+    Return the index (nc) of the grant in the ordered list of grants such that
+    the 95th percentile of the interference from the first nc grants is below the
+    threshold for all azimuths of the receiver antenna.
+
+    Inputs:
+        I:      2D array of interference contributions (dBm/10 MHz); columns
+                correspond to grants, and rows correspond to Monte Carlo iterations
+        t:      95th-percentile protection threshold (dBm/10 MHz)
+
+    Returns:
+        nc:     index nc that defines the move list to be {G_nc+1, G_nc+2, ..., G_Nc}
+    """
+
+    # Create array of incumbent antenna azimuth angles (degrees).
+    azimuths = np.arange(0.0, 360.0, RADAR_BEAMWIDTH/2.0)
+
+    # Initialize nc to Nc.
+    Nc = I.shape[1]
+    nc = Nc
+
+    # Convert protection threshold, interference matrix, and antenna
+    # discrimination to linear units.
+    t_mW = np.power(10.0, t/10.0)
+    I_mW = np.power(10.0, I/10.0)
+    G_discrim = np.power(10.0, INTRF_ATTEN_OUTSIDE_MAIN_BEAM/10.0)
+
+    # Loop through every azimuth angle.
+    for a in azimuths:
+        # Calculate interference contributions at output of receiver antenna.
+        G = np.full((1,Nc),G_discrim)
+
+        idx = grantsInMainBeam(a,bearing)
+        if idx.size:
+            G[:,idx] = 1.0
+        IG = I_mW*G
```

Comment:
All the previous can be replaced using function `antenna.GetRadarNormalizedAntenna()` which returns directly the gains in dB:
```
for azi in azimuths:
  dpa_gains = antenna.GetRadarNormalizedAntennaGains(bearings, azi)
  IG = I_mW  *  10**(dpa_gains/10.)
```